Sources/Chat/ProtocolServices/Invitee/InvitationHandlingService.swift

```diff
@@ -73,47 +70,36 @@ class InvitationHandlingService {
     }
 
     func reject(inviteId: String) async throws {
-
         guard let payload = try invitePayloadStore.get(key: inviteId) else { throw Error.inviteForIdNotFound }
 
-        guard case .invite(let invite) = payload.request.params else {return}
+        guard let requestId = payload.request.id, let invite = try? payload.request.params?.get(Invite.self)
+        else { return }
 
-        let responseTopic = try getInviteResponseTopic(payload, invite)
+        let responseTopic = try getInviteResponseTopic(requestTopic: payload.topic, invite: invite)
 
-        // TODO - error not in specs yet
-        let error = JSONRPCErrorResponse.Error(code: 0, message: "user rejected")
-        let response = JsonRpcResult.error(JSONRPCErrorResponse(id: payload.request.id, error: error))
-
-        try await networkingInteractor.respond(topic: responseTopic, response: response, tag: payload.request.params.responseTag)
+        try await networkingInteractor.respondError(topic: responseTopic, requestId: requestId, tag: Invite.tag, reason: ChatError.userRejected)
 
         invitePayloadStore.delete(forKey: inviteId)
     }
 
     private func setUpRequestHandling() {
-        networkingInteractor.requestPublisher.sink { [unowned self] subscriptionPayload in
-            switch subscriptionPayload.request.params {
-            case .invite(let invite):
-                do {
-                    try handleInvite(invite, subscriptionPayload)
-                } catch {
-                    logger.debug("Did not handle invite, error: \(error)")
-                }
-            default:
-                return
-            }
-        }.store(in: &publishers)
-    }
+        networkingInteractor.requestPublisher.sink { [unowned self] payload in
+            guard payload.request.method == "wc_chatInvite"
```

Comment:
what do you think if at some point we have an enum with protocol methods? so we do like:
```guard payload.request.method == ChatProtocolMethods.invite.rawValue```

Reply:
I will solve these issue in next PR